lib/inspec/metadata.rb

```diff
@@ -212,7 +219,9 @@ def self.finalize_name(metadata, profile_id, original_target)
       return unless metadata.params[:title].nil?
 
       # create a new name based on the original target if it exists
-      metadata.params[:name] = "tests from #{original_target}" unless original_target.to_s.empty?
+      # Crudely slug the target to not contain slashes, to avoid breaking
+      # unit tests that look for warning sequences
+      metadata.params[:name] = "tests from #{original_target}".gsub(%r{[\\\/]}, '.') unless original_target.to_s.empty?
```

Comment:
What do you think about using hyphens instead of periods? That's a more common word separator when slugging a URL, based on absolutely zero evidence I have to provide to you except my gut instinct and some googling. :)

I also wonder if this change is necessary... should we put this line in the `title` metadata instead and put something else in the `name`?

Reply:
I was debating between dashes, underscores, and "something else shell-safe" (I settled on a period, as you see).  My concern with either dashes or underscores is that they're both common in both directory names and folder names, so slugging would lose the path separator....
```
my/profiles/from-steve/no-shenanigans/inspec.yml =>
my-profiles-from-steve-no-shenanigans
```
Maybe that's fine.  

Placing it (unslugged, as originally generated) in `title` works fine.  What should we put in `name` ?  "default" ?

Reply:
I'm pretty agnostic to the hyphens vs periods. I'll let someone else weigh in on that.

I guess what I'm coming back to on this particular change is if this is even necessary since this line is only hit when we're dealing with a file of tests outside of the construct of a profile. I suspect it's not even needed, but given your comment about wishing to potentially catch this in unit tests, I'm totally fine as-is.

Reply:
Since the CLI output will look funky when we strip the slashes, and profiles need to have names, how about we put the actual path as-is in the `title` so that way it at least shows in the output unaltered?